packages/studio-be/src/studio/nlu/nlu-service.ts

```diff
@@ -163,6 +163,29 @@ export class NLUService {
     void bot.train(language)
   }
 
+  public async downloadAndSaveModelWeights(botId: string) {
+    const bot = this._bots[botId]
+    if (!bot) {
+      throw new BotNotMountedError(botId)
+    }
+
+    const botConfig = await this.configProvider.getBotConfig(botId)
+
+    if (!botConfig.nluModels) {
+      throw new Error('Missing NLU models. Bot is not trained.')
+    }
+
+    const modelsFolder = 'models'
+    await this.ghost.forBot(botId).deleteFolder(modelsFolder)
+
+    for (const lang of Object.keys(botConfig.nluModels)) {
+      const modelId = botConfig.nluModels[lang]
+      const modelWeights = await bot.downloadModelWeights(botId, modelId)
+
+      await this.ghost.forBot(botId).upsertFile(modelsFolder, `${modelId}.model`, modelWeights)
```

Comment:
We could probably skip the part where we save the models locally, but this would require multiple refactors, so that's why I opted for the easiest path.

Reply:
:+1: